### PR TITLE
feat: create config dynamically

### PR DESCRIPTION
**Pull request by roman-sainchuk**
## What/Why/How?

In case we are running `openapi-core` in environments not having access to file system we have to be able to create config from string or object.

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

File: packages/core/src/config/load.ts [hunk 1]
```diff
@@ -102,3 +108,21 @@ export async function getConfig(configPath: string | undefined = findConfig()):
...
+export async function createConfig (rawConfig: any, options: CreateConfigOptions = {}): Promise<Config> {
+  if (typeof rawConfig === 'string') {
+    rawConfig = parseYaml(rawConfig);
+  }
+
+  return addConfigMetadata({
+    rawConfig: transformConfig(rawConfig),
+    ...options
+  });
+}
```
Comment (tatomyr) [inline packages/core/src/config/load.ts:128]: I'd avoid overloading.

Maybe union types would work better?

```ts
export async function createConfig (rawConfig: rawConfig | string, options: CreateConfigOptions = {}): Promise<Config> {
 ...
}


```

It's shorter and it's JavaScript.

Comment (roman-sainchuk) [reply packages/core/src/config/load.ts:128]: agree

File: packages/core/src/config/load.ts [hunk 2]
```diff
@@ -111,18 +111,19 @@ export async function getConfig(configPath: string | undefined = findConfig()):
...
-export async function createConfig (rawConfig: RawConfig, options?: CreateConfigOptions): Promise<Config>;
-export async function createConfig (rawConfig: any, options: CreateConfigOptions = {}): Promise<Config> {
-  if (typeof rawConfig === 'string') {
-    rawConfig = parseYaml(rawConfig);
+export async function createConfig(
+  config: string | RawConfig,
+  options?: CreateConfigOptions
+): Promise<Config> {
+  if (typeof config === 'string') {
+    config = parseYaml(config) as RawConfig;
```
Comment (tatomyr) [inline packages/core/src/config/load.ts:122]: Can you avoid mutating the `config` argument?

Comment (roman-sainchuk) [reply packages/core/src/config/load.ts]: Changed

File: packages/core/src/config/load.ts (hunk 1 above)
Comment (roman-sainchuk) [reply packages/core/src/config/load.ts]: Changed

File: packages/core/src/config/load.ts [hunk 3]
```diff
@@ -102,3 +108,20 @@ export async function getConfig(configPath: string | undefined = findConfig()):
     throw new Error(`Error parsing config file at '${configPath}': ${e.message}`);
   }
 }
+
+interface CreateConfigOptions {
```
Comment (tatomyr) [inline packages/core/src/config/load.ts:112]: We usually use `type` notation as more general.

```suggestion
type CreateConfigOptions = {
```

**Review by tatomyr — APPROVED** [review]

**Pull request merged**